compiler/rustc_parse/src/parser/attr_wrapper.rs

```diff
@@ -362,7 +362,10 @@ impl<'a> Parser<'a> {
             let start_pos = if has_outer_attrs { attrs.start_pos } else { start_pos };
             let new_tokens = vec![(FlatToken::AttrTarget(attr_data), Spacing::Alone)];
 
-            assert!(!self.break_last_token, "Should not have unglued last token with cfg attr");
+            if !(self.token.kind == TokenKind::Gt && self.unmatched_angle_bracket_count > 0) {
+                assert!(!self.break_last_token, "Should not have unglued last token with cfg attr");
+            }
```

Comment:
@petrochenkov this change is so that we no longer ICE on the face of

```rust
struct S<#[cfg(feature = "alloc")] N: A<T>>;
                                         ^^ these currently get split and cause the assertion to fail unconditionally
```

I believe that this change is reasonable. Is there *any* reason we shouldn't land this PR as is? Put another way, are there any non-local knock down effects from this change that come to mind?

Reply:
gentle ping @petrochenkov for this comment. thanks.